From 2649696903b87d27d9f9e722f17c7bbe363cea9c Mon Sep 17 00:00:00 2001
From: chdugw_auto <jkulkarni@maxlinear.com>
Date: Thu, 2 Sep 2021 09:55:19 +0300
Subject: [PATCH] Merge pull request #3474 in SW_UGW/linux from
 feature/UGW_SW-61342-dcdp-multicast-gid-get-api-8.x to master

* commit '34ee85e4262dbd9b71aeec59ad25bade82fbca16':
  UGW_SW-61342 : dcdp api to get mcast gid
---
 include/net/directconnect_dp_api.h | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/include/net/directconnect_dp_api.h b/include/net/directconnect_dp_api.h
index b24b37180e45..dcfc383585a1 100644
--- a/include/net/directconnect_dp_api.h
+++ b/include/net/directconnect_dp_api.h
@@ -52,7 +52,7 @@
 /**
   \brief DC DP API version code
  */
-#define DC_DP_API_VERSION_CODE        0x050006
+#define DC_DP_API_VERSION_CODE        0x050007
 
 /**
   \brief DC DP API version
@@ -1852,6 +1852,15 @@ dc_dp_register_mcast_module (
 	uint32_t flags
 );
 
+/** \brief  Get MCAST GID from network packet structure
+  \param[in] skb  pointer to network packet/sk_buff structure
+  \return MCGID >=0 if OK / <0 if error
+ */
+int32_t
+dc_dp_get_skb_gid (
+	struct sk_buff *skb
+);
+
 /** \brief  Provide a Priority (802.1D Priority) to Device QoS/WMM Class/TID map for the
  * given WiFi Radio/net_device
   \param[in] port_id  Port Id on which mapping is to be updated
